### DISCUSSION TITLE
Live Sandbox failing with 'URLSearchParams is not a constructor' and module problem

### DISCUSSION THREAD

**Discussion by o-ja** [gadicc/yahoo-finance2#332] [General]
Hi, 

1) I see that the live sandbox has error at the moment (without modifing anything): https://codesandbox.io/s/yahoo-finance2-312x2?file=/src/index.ts

> > {
>   "name": "TypeError",
>   "message": "URLSearchParams is not a constructor"
> }

2) I also have problems with autoc module with basic example given of querying "GOO". Returns error:

> {
>     "message": "invalid json response body at https://autoc.finance.yahoo.com/autoc?region=1&lang=en&query=GOO reason: Unexpected token < in JSON at position 0",
>     "type": "invalid-json"
> }

**Comment by o-ja** [comment]
Am I missing something or are this actual bugs? If bugs, happy to file than as actual issues.

**Comment by o-ja** [comment]
I have the problems with both yahoo-finance2 versions 1.13.1 and latest 2.0.1.

**Comment by gadicc** [comment]
Thanks, @o-ja.  Confirming the issue.  Will need a bit of time to investigate further.  Thanks for reporting.

**Reply by o-ja** [reply]
Thanks a lot! Monitoring the progress. 🙏

**Comment by gadicc** [comment]
Thanks, @o-ja.  I believe this is working now.  Please confirm when you have a chance.

**Reply by o-ja** [reply]
@gadicc, I see that 1) live sandbox fixed as it returns correctly the price info. Thanks 🙏  

For 2) autoc module. Still I received error when running the following example code given in the docs (other than query should not be required): 

```
import yahooFinance from "yahoo-finance2";
import { show } from "./utils";

show(async () => {
  const query = "Goo";
  const result = await yahooFinance.autoc(query);

  // 2. Put a "." after result to explore
  return result;

  // 3. Try change "quoteSummary" above to something else
});
```

Error:

```
{
  "name": "SyntaxError",
  "message": "Unexpected token < in JSON at position 0"
}
```

Maybe I'm doing something wrong? Let me know.

**Reply by gadicc** [reply]
Ooh sorry, forgot to come back to this.  I had a look now and it seems Yahoo have just totally erased their `autoc` server.  The autocomplete on Yahoo Finance uses `search` under the hood anyways, so check that out, and I've opened https://github.com/gadicc/node-yahoo-finance2/issues/337 for us to update the docs about this.  Thanks for reporting!

**Reply by o-ja** [reply]
Ooooh, nice one! I'll move to the search one. Thanks for letting me know. 👏  👏 

Great work, by the way, on the whole library. I have been enjoying it a lot. 💪

**Reply by gadicc** [reply]
Thanks for your kind words, @o-ja!  I may have started the project but we've had some incredible contributions from a great many people, you can see them all at the bottom of the README :)  Reports like yours also go a long way in keeping up to date with Yahoo's (basically internal) API which is constantly changing, so thanks for all your great contributions too :pray:

**Reply by pgupta1795** [reply]
I am facing this error while running yahooFinance.search, any ideas on the root cause?
